docs/CODE_GUIDELINES.md

```diff
@@ -0,0 +1,341 @@
+
+# Code Guidelines
+
+## Table of Contents
+1. **[Motivation](#1-motivation)**
+2. **[Document Conventions](#2-document-conventions)**
+3. **[Indentation](#3-indentation)**
+4. **[Statements](#4-statements)**
+5. **[Namespaces](#5-namespaces)**
+6. **[Headers](#6-headers)**
+7. **[Braces](#7-braces)**
+8. **[Whitespace](#8-whitespace)**
+9. **[Vertical Alignment](#9-vertical-alignment)**
+10. **[Controls Statements](#10-controls-statements)**  
+  10.1. **[if else](#101-if-else)**  
+  10.2. **[switch case](#102-switch-case)**
+11. **[Naming](#11-naming)**  
+  11.1. **[Namespaces](#111-namespaces)**  
+  11.2. **[Constants](#112-constants)**  
+  11.3. **[Enums](#113-enums)**  
+  11.4. **[Interfaces](#114-interfaces)**  
+  11.5. **[Classes](#115-classes)**  
+  11.6. **[Methods](#116-methods)**  
+  11.7. **[Variables](#117-variables)**
+12. **[Conventions](#12-conventions)**  
+  12.1. **[Casts](#121-casts)**  
+  12.2. **[NULL vs nullptr](#122-null-vs-nullptr)**  
+  12.3. **[auto](#123-auto)**  
+  12.4. **[for loops](#124-for-loops)**
+
+## 1. Motivation
+When working in a large group, the two most important values are readability and maintainability. We code for other people, not computers. To accomplish these goals, we have created a unified set of code conventions.
+
+Conventions can be bent or broken in the interest of making code more readable and maintainable. However, if you submit a patch that contains excessive style conflicts, you may be asked to improve your code before your pull request is reviewed.
+
+## 2. Document Conventions
+Several different strategies are used to draw your attention to certain pieces of information. In order of how critical the information is, these items are marked as a note, tip, or warning. For example: 
+ 
+**NOTE:** Linux is user friendly... It's just very particular about who its friends are.  
+**TIP:** Algorithm is what developers call code they do not want to explain.  
+**WARNING:** Developers don't change light bulbs. It's a hardware problem.
+
+**[back to top](#table-of-contents)**
+
+## 3. Indentation
+Use spaces as tab policy with an indentation size of 2.
+
+**[back to top](#table-of-contents)**
+
+## 4. Statements
+No multiple statements on a single line.
+```cpp
+std::vector<std::string> test; test.push_back("foobar");
+```
+Always use a new line for a new statement.
+```cpp
+std::vector<std::string> test;
+test.push_back("foobar");
+```
+It is much easier to debug if one can pinpoint a precise line number.
+
+**[back to top](#table-of-contents)**
+
+## 5. Namespaces
+Indentation is not required to simplify nested namespaces and wrapping *.cpp* files in a namespace.
+```cpp
+namespace KODI
+{
+namespace UTILS
+{
+class ILogger
+{
+  void Log(...) = 0;
+}
+}
+}
+```
+
+**[back to top](#table-of-contents)**
+
+## 6. Headers
+Included header files should be sorted alphabetically to prevent duplicates and allow better overview, with an empty line clearly separating sections.
```

Comment:
Included header files have to be sorted ...

Reply:
Fixed